waspc/src/Wasp/Generator/Common.hs

```diff
@@ -38,3 +42,16 @@ npmVersionRange =
 
 prismaVersion :: SV.Version
 prismaVersion = SV.Version 3 15 2
+
+-- | Adds .cmd extension to callable npm command if ran on Windows, since npm does note have .exe on Windows.
+--   The reason explained here: https://stackoverflow.com/questions/43139364/createprocess-weird-behavior-with-files-without-extension
+oSSpecificNpm :: String
+oSSpecificNpm = "npm" ++ if os /= "mingw32" then "" else ".cmd"
```

Comment:
I usually try to document as much as I can with the code, but if I can't capture it in the code, then it goes in the comments, with as little duplication as possible.

Also, I differ between comments explaining functions API (what code calling it needs to know about it) and comments explaining implementation details (what developer maintaining the innards of the function needs to know about it).

So in this case, I would do some modifications like this:
```suggestion
npmCmd :: String
npmCmd =
  -- Windows add ".exe" to command, when calling it programmatically, if it doesn't
  -- have an extension already, meaning that calling `npm` actually calls `npm.exe`.
  -- However, there is no `npm.exe` on Windows, instead there is `npm` or `npm.cmd`, so we make sure here to call `npm.cmd`.
  -- Extra info: https://stackoverflow.com/questions/43139364/createprocess-weird-behavior-with-files-without-extension .
  "npm" ++ if os /= "mingw32" then "" else ".cmd"
```